### ISSUE TITLE
`TypeError` on `import pykep as pk`

### ISSUE THREAD

**Issue by peddie** [esa/pykep#77]
I've installed pykep release 2.2 from source on Debian linux, x86_64 according to [the instructions](https://esa.github.io/pykep/installation.html).  I have pagmo and pygmo installed and can run examples in my python interpreter.  When I try to test my installation by running [the first example](https://esa.github.io/pykep/examples/ex1.html), I get the following error:

```
peddie@jaidee:build(tags/v2.2☠)$ ipython
Python 2.7.15 (default, May  1 2018, 05:55:50) 
Type "copyright", "credits" or "license" for more information.

IPython 5.5.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import pykep as pk
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-e89771118bdc> in <module>()
----> 1 import pykep as pk

/usr/local/lib/python2.7/dist-packages/pykep/__init__.pyc in <module>()
     46 # importing *.
     47 from pykep.core import *
---> 48 from pykep import core, sims_flanagan, pontryagin, orbit_plots, examples, phasing, util, planet, trajopt
     49 
     50 

/usr/local/lib/python2.7/dist-packages/pykep/trajopt/__init__.py in <module>()
     15     from pykep.trajopt._pl2pl_N_impulses import pl2pl_N_impulses
     16     # The open traj gym
---> 17     from pykep.trajopt._gym import gym
     18     gym = gym()
     19     # The launchers models

/usr/local/lib/python2.7/dist-packages/pykep/trajopt/_gym.py in <module>()
     86         return self.get_name()
     87 
---> 88 class gym:
     89     def __init__(self):
     90         pass

/usr/local/lib/python2.7/dist-packages/pykep/trajopt/_gym.py in gym()
     90         pass
     91     # Problem P1: Cassini MGA, single objective, direct encoding
---> 92     cassini1 = _cassini1_udp()
     93     # Problem P2: Cassini MGA, single objective, alpha encoding
     94     cassini1a =_cassini1a_udp()

/usr/local/lib/python2.7/dist-packages/pykep/trajopt/_gym.py in __init__(self)
      7 class _cassini1_udp(mga):
      8     def __init__(self):
----> 9         super(_cassini1_udp, self).__init__(
     10             seq=_seq_cassini,
     11             t0=[-1000., 0.],

TypeError: super() argument 1 must be type, not classobj
```

The error is identical (without source context) if I use the vanilla python interpreter in place of ipython.  `cmake` seems to be correctly detecting my python version and so on:

```
peddie@jaidee:build(tags/v2.2☠)$ cmake ..
-- System name: Linux
-- The C++ compiler ID is: GNU
-- YACMA autodetected C++ flags: -fdiagnostics-color=auto;-ftemplate-depth=1024;-fvisibility-inlines-hidden;-fvisibility=hidden;-Wno-attributes
-- YACMA autodetected C++ debug flags: -Wall;-Wextra;-Wnon-virtual-dtor;-Wnoexcept;-Wlogical-op;-Wconversion;-Wdeprecated;-pedantic-errors;-Wdisabled-optimization;-ftemplate-backtrace-limit=0;-fstack-protector-all;-Wodr;-Wsuggest-final-types;-Wsuggest-final-methods;-Wshift-negative-value;-Wshift-overflow=2;-Wduplicated-cond;-Wrestrict;-Waligned-new
-- Python modules do NOT require linking to the Python library.
-- Python interpreter: /usr/bin/python
-- Python interpreter version: 2.7.15
-- Python include dir: /usr/include/python2.7
-- Generic UNIX platform detected.
-- Python modules install path: lib/python2.7/dist-packages
-- Boost version: 1.62.0
-- Found the following Boost libraries:
--   date_time
--   serialization
-- Required Boost libraries: date_time;serialization;python
-- Boost version: 1.62.0
-- Found the following Boost libraries:
--   date_time
--   serialization
--   python
-- Detected Boost version: 106200
-- Boost include dirs: /usr/include
-- Setting up the compilation of the Python module '_core'.
-- Setting up extra compiler flag '-fwrapv' for the Python module '_core'.
-- Python < 3 detected, setting up extra compiler flag '-fno-strict-aliasing' for the Python module '_core'.
-- Setting up the compilation of the Python module '_planet'.
-- Setting up extra compiler flag '-fwrapv' for the Python module '_planet'.
-- Python < 3 detected, setting up extra compiler flag '-fno-strict-aliasing' for the Python module '_planet'.
-- Setting up the compilation of the Python module '_sims_flanagan'.
-- Setting up extra compiler flag '-fwrapv' for the Python module '_sims_flanagan'.
-- Python < 3 detected, setting up extra compiler flag '-fno-strict-aliasing' for the Python module '_sims_flanagan'.
-- Setting up the compilation of the Python module '_util'.
-- Setting up extra compiler flag '-fwrapv' for the Python module '_util'.
-- Python < 3 detected, setting up extra compiler flag '-fno-strict-aliasing' for the Python module '_util'.
-- Creating the files for the generation of a binary wheel.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/peddie/software/numerical/pykep/build
```

What am I doing wrong?  What must I do to run the examples?

Happy to provide any additional information that can help debug.

**Comment by darioizzo** [comment]
This is a py27 issue with the syntax we use (https://stackoverflow.com/questions/1713038/super-fails-with-error-typeerror-argument-1-must-be-type-not-classobj-when). It is very strange though that our continuous integration does not catch the issue. See (https://travis-ci.org/esa/pykep/jobs/400856266#L2888) for a successful build and import in a linux system.

* Why are you compiling? pip does not work for you?
* A work around for you is to just change the class syntax to the old style class my_class(object)

Let me know

**Comment by peddie** [comment]
Thanks for your reply.

> Why are you compiling? pip does not work for you?

I decided to compile when the pip package gave the same error!  I reported the results from the direct installation from source because I thought I was removing a variable from the equation :)  Sorry for the confusion.

> A work around for you is to just change the class syntax to the old style class my_class(object)

I'm afraid I'm not familiar with how python's object system works or has changed recently.  Could you be a bit more specific about the fix you're recommending?  Are you suggesting I change
```
class _cassini1_udp(mga):
```
to
```
class _cassini1_udp(mga, object):
```
or should I be adding `object` to the declaration of `class mga:` in `trajopt/_mga.py`?

**Comment by peddie** [comment]
I tried changing `class mga:` to `class mga(object):` in `trajopt/_mga.py`, but I encountered a new error:
```
peddie@jaidee:~$ python
Python 2.7.15 (default, May  1 2018, 05:55:50) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pykep as pk
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/peddie/.local/lib/python2.7/site-packages/pykep/__init__.py", line 48, in <module>
    from pykep import core, sims_flanagan, pontryagin, orbit_plots, examples, phasing, util, planet, trajopt
  File "/home/peddie/.local/lib/python2.7/site-packages/pykep/trajopt/__init__.py", line 27, in <module>
    from pykep.trajopt._direct import direct_pl2pl
  File "/home/peddie/.local/lib/python2.7/site-packages/pykep/trajopt/_direct.py", line 452
    self.Eflb, *(-1, -1, -1) * self.nseg]
               ^
SyntaxError: invalid syntax
```

Have I simply got the wrong version of python?  What version do you mainly use to develop?  I don't use python for anything else, so I can happily change to whatever version is likely to be the best-tested.

**Comment by darioizzo** [comment]
if you use python 3.6 everything should be fine.
It is still very strange that you get this as our py27 system import pykep no problem ...

**Comment by ghorn** [comment]
You could just use python 3.6

**Comment by peddie** [comment]
@darioizzo, I've got the first example to run successfully under 3.6 now, after some faffing about with pagmo settings (I have no SNOPT but have a custom IPOPT), and am able to try my own problems.  Is there any other information I can give you to help understand where the error is coming from under Python 2.7?  The git commit I was using when I encountered the error was [`abde4754f1d0ca24abab2953a809097f04658ef6`](https://github.com/esa/pykep/commit/abde4754f1d0ca24abab2953a809097f04658ef6).
